[Lau Teng Hon] iP

Th-429B opened the pull request:
# DukeNotPro
> "The more I C, the less I see."
> -Annoymous

DukeNotPro came at the cost of many hours and sleep. It's
- definitely not text-based
- easy to use(?)
- ~~FAST!~~ Super Fast... maybe 

Follow these instructions to download:
1. download it from [here](https://www.youtube.com/watch?v=6n3pFFPSlW4)
2. look at 4.
3. there's no 4 :rofl:
**teehee**

Features:
- [x] Managing tasks.
- [ ] GUI (coming soon)

Now for some random filler code here's the ```main``` method.
```java
public class Main {
    public static void main(String[] args) {
        System.out.print("Have a nice day! Take a break sometimes.");
    }
}
```

Commit:
Add Gradle support

Th-429B committed:
add greet, echo and exit functions

Th-429B committed:
add "add" function and "list" function

Th-429B committed:
Add done function

Th-429B committed:
Cleanup code

Th-429B committed:
Optimise code using switch

Th-429B committed:
Add features for Level-3

Th-429B committed:
Add A-TextUiTesting

Th-429B committed:
Code optimisation

Th-429B committed:
Add exception handling
Done:
check if number is empty
check if the number is a valid number
check if the input is valid
Deadline:
check if description is empty
check if deadline is empty
Todo:
check if task description is empty
Event:
check if event name is empty
check if time is empty

Th-429B committed:
Add level-6 features

Th-429B committed:
Seperate files into their packages

Th-429B committed:
Add level-7 features

Th-429B committed:
Add Level-8 features

Th-429B committed:
Merge branch 'branch-Level-7'

Th-429B committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/duke/ChatBot.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/tasks/Deadline.java
#	src/main/java/duke/tasks/Event.java

Th-429B committed:
Refactor ChatBot.java to extract UI.java & TaskList.java

Th-429B committed:
Refactor ChatBot.java to extract Parser.java

Th-429B committed:
Shift TaskList into tasks folder

Th-429B committed:
Remove obsolete code and unused libraries

Th-429B committed:
Add junit test for Parser and TaskList

Th-429B committed:
Fix file path issue

Th-429B committed:
Add JavaDocs for all java files

Th-429B committed:
Reformat code to follow coding standard

Th-429B committed:
Add Level-9 feature (find)

Th-429B committed:
Merge branch 'branch-Level-9'

Th-429B committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/ChatBot.java

Th-429B committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/tasks/TaskList.java

Th-429B committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

Th-429B committed:
Merge branch 'branch-A-Gradle'

Th-429B committed:
Add checkstyle to gradle

Th-429B committed:
Merge branch 'Branch-A-CheckStyle'

Th-429B committed:
Fix coding violations

Th-429B committed:
Add GUI to Duke

Th-429B committed:
Update UI to remove the line for CLI

Th-429B committed:
Merge branch 'Branch-Level-10'

Th-429B committed:
Add javadocs for new methods

Th-429B committed:
Refactor code into gui package

Th-429B committed:
Remove unused code

Th-429B committed:
Fix bye command not exiting the program

Th-429B committed:
Cleanup unused params

loose-bus-change commented on src/main/java/duke/ChatBot.java:
```diff
@@ -0,0 +1,170 @@
+ */
+public class ChatBot {
+
+    private ArrayList<Task> temp;
+    private TaskList tasklist;
```
Are these two collections ?

Th-429B replied:
Yes they are, just that TaskList is an abstraction of the collection of tasks.

loose-bus-change commented on src/main/java/duke/ChatBot.java:
```diff
@@ -0,0 +1,170 @@
+    }
+
+    /**
+     * This method is to add a new deadline to the storage and
+     * print out the add message when the task is added successfully.
```
Personally I like this header, its very clear and neat! You could also consider just beginning the description with 'returns' or 'adds' or 'sends' etc. So for instance, 'adds a new deadline ....'.

Th-429B replied:
Will update the javadocs to fit documentation standards

loose-bus-change commented on src/main/java/duke/ChatBot.java:
```diff
@@ -0,0 +1,170 @@
+    public String handleDeadline(String name, LocalDateTime deadline) {
+        Task t = tasklist.addDeadline(name, deadline, false);
+        s.saveTasks(tasklist);
+        return ui.showAddTaskMessage(t, tasklist.getTotalTasksNumber());
```
I like the names of these methods, very descriptive and clear!

loose-bus-change commented on src/main/java/duke/ChatBot.java:
```diff
@@ -0,0 +1,170 @@
+
+    /**
+     * This method gets the total number of tasks.
+     * @return total number of tasks.
```
I think there should be a blank line between the description and the param/return

Th-429B replied:
Noted, will make the changes

loose-bus-change reviewed (commented):
LGTM!

Th-429B committed:
Update gradle build file

Main class in current gradle build file is not updated. Any .jar builds
from that version will not run properly.

MainClassName is updated to duke.Launcher.

Launcher has been updated to be the new main class in duke.

Th-429B committed:
Add assertions to TaskList

There is no checking of valid index passed to completeTask and
deleteTask.

Although there are prechecks before the method is called,
should those checks be removed, we can prevent the program from crashing
using assert in TaskList.

Added assertions in TaskList to assert that index cannot be negative or
bigger than TaskList size.

Th-429B committed:
Refactor javadocs to fit convention

Current javadocs do not fit with convention, starting the javadocs with
returns, adds etc...

Change javadocs to follow convention so that reviewers will not get
confused.

Th-429B committed:
Refactor: Replace Magic Literal

The code uses int 0 and 1 to denote when to terminate.

Replace the magic literals with an enum of START and STOP.

Since exit status is a constant enum was used.

Th-429B committed:
Refactor: Extract Method

Current code has multiple indirect checks for string array length and
valid index.

Extract methods to check for string array length and valid index.

Extracted methods improve code readability and follows SLAP (single
level of abstraction principle.

Th-429B committed:
Fix error when user uses alphabets to specify task number to be
delete/completed.

There was no error checking for such a situation.

Throws an exeception when encountering such situation.

To account for number format exeception.

Th-429B committed:
Merge pull request #1 from Th-429B/Branch-A-Assertions

Add assertions to TaskList

Th-429B committed:
Merge pull request #2 from Th-429B/Branch-A-CodeQuality

Branch a code quality

Th-429B committed:
Set up Continuous Integration (CI).

Th-429B committed:
Add sort command

Current code does not allow users to sort their tasks when using "list"
command

Added a new feature "sort"

Added two classes that implements comparator since we might want to sort
tasks by multiple parameters.

Th-429B committed:
Update user and duke profile picture

Th-429B committed:
Fix Storage bug

Storage was not properly creating the folders required.

Added the code to make the folders.

Storage now works when build to jar.

Th-429B committed:
Added representative screenshot

Th-429B committed:
Updated README

Th-429B committed:
Set theme jekyll-theme-slate

Th-429B committed:
Set theme jekyll-theme-midnight

Th-429B committed:
Update README.md

Fix styling issue

Th-429B committed:
Delete UI.png.png

Th-429B committed:
Fixed naming issue

Th-429B committed:
Merge branch 'master' of https://github.com/Th-429B/ip

# Conflicts:
#	docs/Ui.png.png

Th-429B committed:
Fix naming issue

Th-429B committed:
Fix coding violations

Th-429B committed:
Fix bug in ParserTest

execute_doneInvalidNumber_execeptionThrown() was getting null pointer
exception.

Did not initialize chatBot.

Added line to initialize chatBot, bot.Start().

Th-429B committed:
Change Duke to TaskMe

Change TaskMe corresponding UI response

Th-429B committed:
Update README.md

Th-429B committed:
Update Ui.png

Th-429B committed:
Merge branch 'master' of https://github.com/Th-429B/ip

Th-429B committed:
Update README.md

Add feature list to README

Th-429B committed:
Update README.md

fix screenshot of starting message not loading

Th-429B committed:
Update README.md

fix formatting issues

Th-429B committed:
Update README.md

Th-429B committed:
Update README.md

Th-429B committed:
Update README.md
